docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,198 @@
+
+***
+
+![Diagram of the audit v2 system](images/audit-v2-design.svg)
+
+***
+
+
+#### accounting
+The accounting binary is responsible for calculating invoices for uplinks and payments for storage nodes. In order to do this, accounting must track total amounts of disk usage and bandwidth used by storage nodes and by buckets. Accounting should receive storage node total stored bytes data from the tally observer running with the metainfo loop.
+
+#### version
+Can we remove the version service? Or do we want to keep it and let these separated binaries check in with the version service to check they are running an appropriate version? Not sure if that is necessary since we could handle version compatibility at the layer of infrastructure deployment.
+
+#### uptime ?
+Is uptime going to be a system?
+
+***
+
+The following diagram shows the above propsed design:
+
+![Diagram of the above listed binaries](images/sa-separation-design.svg)
+
+***
+
+## Rationale
+
+#### metainfo loop and observer system
+For database performance reasons we should only have one thing looping over the metainfo database. This is why we are combining everything into the metainfo loop observer system. There is no sense in having multiple different processes all redoing the work of iterating over the database and parsing protobufs if lots of things need to do that same work over and over. Even with the observers running on different intervals, we still thinks its fine to combine all the 
+
+There has been discussion about making the audit and GC observers run on differnt loops, but for performance concerns, its been decided to run all observers on the same metainfo loop (though GC will run less frequently). 
+
+An additional is that for each metainfoDB iteration, we might want to choose a random starting position
+so that the reservoir samples for the audit observer aren't biased toward the items at the beginning of the metainfoDB. If we start at a random location we can still loop through everything, we just have to make sure to wrap back to the beginning when we reach the end and keep going until we get back to the starting place. Keep in mind it's important for the GC observer that we hit every single item in the metainfoDB otherwise we could mess up GC output if we don't get an accurate list of current pieceIDs for storage nodes.
+
+For the metainfo loop and observer system its not critical to have high availability for these systems, therefore its ok to have all these observers depend on a single metainfo loop process.  In the worst case scenario, if the metainfo loop goes down there may be downtime until its fixed and back up, this should be fine for small periods of time.
+
+#### satellite api
+
+We chose to combine all public GRPC and HTTP endpoints into one binary. The reason for this is so that we handle all requests in one place so we can scale handling requests up easier.
```

Comment:
The piece which handles this will (I would guess) be responsible for passing requests on to the appropriate backend systems, and load-balancing between them if there are multiple candidates.